File: codingTest/baekjoon/python/1260.py
from collections import deque


def dfs(v):
    if visited[v]:
        return
    visited[v] = True
    print(v, end=" ")
    for i in arr[v]:
        if not visited[i]:
            dfs(i)


def bfs(v):
    q = deque([v])
    while q:
        node = q.popleft()
        if visited[node]:
            continue
        print(node, end=" ")
        visited[node] = True
        for i in arr[node]:
            q.append(i)


n, m, v = map(int, input().split())
arr = [[] for _ in range(n + 1)]
for _ in range(m):
    x, y = map(int, input().split())
    arr[x].append(y)
    arr[y].append(x)
for i in arr:
    i.sort()  # 정점 번호가 작은 것 부터

visited = [False for _ in range(n + 1)]
dfs(v)
print()
visited = [False for _ in range(n + 1)]
bfs(v)

File: codingTest/baekjoon/javascript/1260.js
const rl = require("readline").createInterface({ input: process.stdin, output: process.stdout });

const input = [];
rl.on("line", line => {
  input.push(line);
}).on("close", () => {
  var [n, m, v] = input
    .shift()
    .split(" ")
    .map(e => +e);
  var arr = Array.from(Array(n + 1), () => []);
  input.some(e => {
    var [x, y] = e.split(" ").map(e => +e);
    arr[x].push(y);
    arr[y].push(x);
  });
  for (let i of arr) i.sort((a, b) => a - b);
  let visited = new Array(n + 1).fill(false);
  let dfsResult = "";
  let bfsResult = "";
  const dfs = v => {
    if (visited[v]) return;
    visited[v] = true;
    dfsResult += v + " ";
    for (let i of arr[v]) dfs(i);
  };
  const bfs = v => {
    let q = [v];
    while (q.length) {
      let item = q.shift();
      if (visited[item]) continue;
      visited[item] = true;
      bfsResult += item + " ";
      for (let i of arr[item]) q.push(i);
    }
  };
  dfs(v);
  visited = new Array(n + 1).fill(false);
  bfs(v);
  console.log(dfsResult);
  console.log(bfsResult);
});
